plugins/inputs/radius/radius.go

```diff
@@ -0,0 +1,118 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package radius
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"layeh.com/radius"
+	"layeh.com/radius/rfc2865"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Radius struct {
+	Servers         []string        `toml:"servers"`
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Secret          config.Secret   `toml:"secret"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	Log             telegraf.Logger `toml:"-"`
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (r *Radius) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *Radius) Init() error {
+	if len(r.Servers) == 0 {
+		r.Servers = []string{"127.0.0.1:1812"}
+	}
+	if r.ResponseTimeout < config.Duration(time.Second) {
+		r.ResponseTimeout = config.Duration(time.Second * 5)
+	}
+	return nil
+}
+
+func (r *Radius) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, server := range r.Servers {
+		wg.Add(1)
+		go func(server string) {
+			defer wg.Done()
+			acc.AddError(r.pollServer(server, acc))
+		}(server)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (r *Radius) pollServer(server string, acc telegraf.Accumulator) error {
+	// Create the fields for this metric
+	host, port, err := net.SplitHostPort(server)
+	tags := map[string]string{"source": host, "port": port}
+	fields := make(map[string]interface{})
+
+	// Create the radius Client
+	client := &radius.Client{
+		Retry: 0,
+	}
+
+	secret, err := r.Secret.Get()
+	if err != nil {
+		return fmt.Errorf("getting secret failed: %v", err)
+	}
+	username, err := r.Username.Get()
+	if err != nil {
+		return fmt.Errorf("getting username failed: %v", err)
+	}
+	password, err := r.Password.Get()
+	if err != nil {
+		return fmt.Errorf("getting password failed: %v", err)
+	}
```

Comment:
You need to release the secrets as those are in mlock'd memory...
```suggestion
	secret, err := r.Secret.Get()
	if err != nil {
		return fmt.Errorf("getting secret failed: %v", err)
	}
	defer config.ReleaseSecret(secret)
	
	username, err := r.Username.Get()
	if err != nil {
		return fmt.Errorf("getting username failed: %v", err)
	}
	defer config.ReleaseSecret(username)

	password, err := r.Password.Get()
	if err != nil {
		return fmt.Errorf("getting password failed: %v", err)
	}
	defer config.ReleaseSecret(password)
```